### PR TITLE
4605 rsync upload lock issue

### PR DESCRIPTION
**Pull request by landreev**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #4605: rsync upload lock issue

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

**Comment by pameyer** [comment]
Functionality looks good - locking consistent with UI/API dataset creation and retrieving transfer script.

Some duplications appear to be duplicated (3x notification on checksum verification success, 2x on checksum verification failure) - these might be due to misconfiguration on my part, and might be out of scope for 4605.

File: src/main/java/edu/harvard/iq/dataverse/batch/jobs/importer/filesystem/FileRecordWriter.java [hunk 1]
```diff
@@ -166,6 +167,13 @@ public void writeItems(List list) {
                     jobContext.setExitStatus("FAILED");
                     return;
                 }
+                DatasetLock dcmLock = dataset.getLockFor(DatasetLock.Reason.DcmUpload);
```
Comment (matthew-a-dunlap) [inline src/main/java/edu/harvard/iq/dataverse/batch/jobs/importer/filesystem/FileRecordWriter.java]: Its kind of confusing that a writeItems method is removing a lock and updating the dataset version. is there a reason this doesn't live in the close method?

**Comment by landreev** [comment]
@pameyer - is this looking like this extra notifications are new, as in introduced in this PR?

**Comment by pameyer** [comment]
@landreev - the extra (admin only) notifications aren't new in this PR

File: src/main/java/edu/harvard/iq/dataverse/batch/jobs/importer/filesystem/FileRecordWriter.java (hunk 1 above)
Comment (matthew-a-dunlap) [reply src/main/java/edu/harvard/iq/dataverse/batch/jobs/importer/filesystem/FileRecordWriter.java]: Ahh ok. Thanks for the clarification!

**Pull request merged**